Source/ASCollectionView.mm

```diff
@@ -1831,6 +1831,17 @@ - (void)rangeController:(ASRangeController *)rangeController didUpdateWithChange
       
       __block NSUInteger numberOfUpdates = 0;
       [self _superPerformBatchUpdates:^{
+        
+        // Use native moveItem instead of Delete-Insert
+        NSMutableArray *fromIndexPaths = [NSMutableArray array];
+        NSMutableArray *toIndexPaths = [NSMutableArray array];
+        for (_ASHierarchyMoveItemChange *change in [changeSet itemChangesOfType:_ASHierarchyChangeTypeMove]) {
+          [super moveItemAtIndexPath:change.fromIndexPath toIndexPath:change.toIndexPath];
+          [fromIndexPaths addObject:change.fromIndexPath];
+          [toIndexPaths addObject:change.toIndexPath];
+          numberOfUpdates++;
+        }
+        
```

Comment:
This part may be irrelevant, as @Adlai-Holler was mentioning the change to dataSource, and this part is regarding to the call to UICollectionView.

Consider the comment I made above, after changing the move mechanism to something similar to current reload, we may have a part to do what @Adlai-Holler said.

Appreciate for further discussion.